Content.Server/Fluids/EntitySystems/SpraySystem.cs

```diff
@@ -57,42 +57,41 @@ private void OnAfterInteract(EntityUid uid, SprayComponent component, AfterInter
             return;
         }
 
-        var playerPos = Transform(args.User).Coordinates;
+        var playerMapPos = Transform(args.User).MapPosition;
+        var clickMapPos = args.ClickLocation.ToMap(EntityManager);
 
-        if (args.ClickLocation.GetGridUid(EntityManager) != playerPos.GetGridUid(EntityManager))
+        var diffPos = clickMapPos.Position - playerMapPos.Position;
+        var diffLength = diffPos.Length;
+        if (diffLength == 0f)
             return;
 
-        var direction = (args.ClickLocation.Position - playerPos.Position).Normalized;
-        var threeQuarters = direction * 0.75f;
-        var quarter = direction * 0.25f;
+        var diffNorm = diffPos.Normalized;
+        var threeQuarters = diffNorm * 0.75f;
+        var quarter = diffNorm * 0.25f;
 
         var amount = Math.Max(Math.Min((solution.CurrentVolume / component.TransferAmount).Int(), component.VaporAmount), 1);
 
         var spread = component.VaporSpread / amount;
 
         for (var i = 0; i < amount; i++)
         {
-            var rotation = new Angle(direction.ToAngle() + Angle.FromDegrees(spread * i) -
+            var rotation = new Angle(diffNorm.ToAngle() + Angle.FromDegrees(spread * i) -
                                      Angle.FromDegrees(spread * (amount - 1) / 2));
 
-            var (_, diffPos) = args.ClickLocation - playerPos;
-            var diffNorm = diffPos.Normalized;
-            var diffLength = diffPos.Length;
-
-            var target = Transform(args.User).Coordinates
+            var target = playerMapPos
                 .Offset((diffNorm + rotation.ToVec()).Normalized * diffLength + quarter);
 
-            if (target.TryDistance(EntityManager, playerPos, out var distance) && distance > component.SprayDistance)
-                target = Transform(args.User).Coordinates
-                    .Offset(diffNorm * component.SprayDistance);
+            var distance = target.Position.Length;
+            if (distance > component.SprayDistance)
+                target = playerMapPos.Offset(diffNorm * component.SprayDistance);
 
             var newSolution = _solutionContainerSystem.SplitSolution(uid, solution, component.TransferAmount);
 
             if (newSolution.TotalVolume <= FixedPoint2.Zero)
                 break;
 
             var vapor = Spawn(component.SprayedPrototype,
-                playerPos.Offset(distance < 1 ? quarter : threeQuarters));
+                playerMapPos.Offset(distance < 1 ? quarter : threeQuarters));
             Transform(vapor).LocalRotation = rotation;
```

Comment:
If the vapor is being attached to map / grid should this just be worldrotation being set?